pkg/servicescm/servicescm.go

```diff
@@ -91,15 +93,21 @@ type Data struct {
 	Services []Service `json:"services"`
 	// Files contains the path and checksum of all the files copied to a Windows VM by WMCO
 	Files []FileInfo `json:"files"`
+	// EnvironmentVars will be set as system-level environment variables on each Windows instance
+	EnvironmentVars map[string]string `json:"environmentVars,omitempty"`
 }
 
-// NewData returns a new 'Data' object with the given services and files. Validates given object contents on creation.
-func NewData(services *[]Service, files *[]FileInfo) (*Data, error) {
-	cmData := &Data{*services, *files}
+// NewData returns a new 'Data' object with the given services, files and ENV vars if they exist.
+// Validates given object contents on creation.
+func NewData(services *[]Service, files *[]FileInfo, envVars map[string]string) (*Data, error) {
+	cmData := Data{Services: *services, Files: *files}
+	if envVars != nil {
+		cmData.EnvironmentVars = envVars
+	}
```

Comment:
Is this nil check necessary?
it will be set to nil either way will it not?

Both by being uninitialized with a default of nil or being set to the value of envVars, which is nil.

Reply:
changed+added a comment for clarity